File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function (tokens) {
    if (tokens.length == 1) return Number(tokens[0]);
    let numPipe = [];

    for (let i = 0; i < tokens.length; i++) {
        let temp;
        const value = tokens[i];
        switch (value) {
            case "/":
                temp = numPipe.pop();
                numPipe.push(Math.trunc(numPipe.pop() / temp));
                break;
            case "*":
                temp = numPipe.pop();
                numPipe.push(numPipe.pop() * temp);
                break;
            case "+":
                temp = numPipe.pop();
                numPipe.push(numPipe.pop() + temp);
                break;
            case "-":
                temp = numPipe.pop();
                numPipe.push(numPipe.pop() - temp);
                break;
            default:
                numPipe.push(Number(value));
        }
        
        console.log(numPipe);


    }

    return numPipe.pop();
};

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.cs
public class Solution {
    public int EvalRPN(string[] tokens) {
        if (tokens.Length == 1) return int.Parse(tokens[0]);
        var result = new Stack<int>();

        for(int i = 0; i < tokens.Length; i++){
            string value = tokens[i];
            int helper;
            switch(value){
                case "+":
                    helper = result.Pop();
                    result.Push(result.Pop() + helper);
                    break;
                case "-":
                    helper = result.Pop();
                    result.Push(result.Pop() - helper);
                    break;
                case "*":
                    helper = result.Pop();
                    result.Push(result.Pop() * helper);
                    break;
                case "/":
                    helper = result.Pop();
                    result.Push(result.Pop() / helper);
                    break;
                default:
                    result.Push(int.Parse(value));
                    break;
            }
        }

        return result.Pop();
    }
}